src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketRepositoryClientImpl.java

```diff
@@ -50,6 +66,30 @@ public BitbucketBuildStatusClient getBuildStatusClient(String revisionSha,
         return new BitbucketBuildStatusClientImpl(bitbucketRequestExecutor, revisionSha);
     }
 
+    @Override
+    @SuppressFBWarnings("NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE")
+    public BitbucketBranchClient getBranchClient(TaskListener listener, SCMSourceOwner context, BitbucketSCMRepository repository) throws IOException, InterruptedException {
+        Git git = Git.with(listener, new EnvVars(EnvVars.masterEnvVars));
+        GitClient client = git.getClient();
+
+        if (client == null) {
+            log.log(Level.WARNING, "failed to initialise git client");
+            throw new IOException();
+        }
+
+        BitbucketRepository bitbucketRepository = getRepository();
+        if (repository.getSshCredentialsId() != null && !repository.getSshCredentialsId().isEmpty()) {
```

Comment:
You can use `repository.getCloneCredentialsId`, rather than having to use a conditional here.